extensions/picocli/runtime/src/main/java/io/quarkus/picocli/runtime/PicocliBeansFactory.java

```diff
@@ -0,0 +1,12 @@
+package io.quarkus.picocli.runtime;
+
+import javax.enterprise.inject.spi.CDI;
+
+import picocli.CommandLine;
+
+class PicocliBeansFactory implements CommandLine.IFactory {
+    @Override
+    public <K> K create(Class<K> aClass) {
+        return CDI.current().select(aClass).get();
```

Comment:
It is important that the factory falls back to the picocli default factory: the default factory has logic to instantiate collections (Lists, Sets, Maps) that picocli relies on. How about something like this:

```java
class PicocliBeansFactory implements CommandLine.IFactory {

    private final IFactory defaultFactory = CommandLine.defaultFactory();

    @Override
    public <K> K create(Class<K> aClass) throws Exception {
        Instance<K> instance = CDI.current().select(aClass);
        if (instance.isResolvable()) {
            // question: does Quarkus provide some internal logging facility?
            //System.out.println("Got instance " + instance + " via " + CDI.current());
            return instance.get();
        }
        //System.out.println("Instantiating " + aClass + " via default picocli factory");
        return defaultFactory.create(aClass);
    }
}
```